### PR TITLE
fix: Correctly process subseconds in `pl.duration`

### LINKED ISSUE #11747: `duration` doesn't handle subseconds correctly
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
pl.select(pl.duration(seconds=1, microseconds=1000, time_unit='ms'))
```
```
shape: (1, 1)
┌──────────────┐
│ duration     │
│ ---          │
│ duration[ms] │
╞══════════════╡
│ 1s           │
└──────────────┘
```

### Log output

_No response_

### Issue description

When `time_unit` is milliseconds, all nanosecond/microsecond information is lost.
When `time_unit` is microseconds, all nanosecond information is lost.

### Expected behavior

Information should be preserved.

### Installed versions

main branch

### PR DESCRIPTION
**Pull request by stinodego**
Closes #11747

#### Changes

* Refactor the function and related tests
* Add new test and fix the bug

### CODE REVIEW

File: crates/polars-plan/src/dsl/functions/temporal.rs [hunk 1]
```diff
@@ -256,95 +256,103 @@ impl DurationArgs {
...
+        // Process subseconds
+        let max_len = s.iter().map(|s| s.len()).max().unwrap();
         let mut duration = match args.time_unit {
+            TimeUnit::Microseconds => {
+                if is_scalar(&microseconds) {
+                    microseconds = microseconds.new_from_index(0, max_len);
+                }
+                if !is_zero_scalar(&nanoseconds) {
+                    microseconds = microseconds + (nanoseconds / 1_000);
+                }
```
Comment (stinodego) [inline crates/polars-plan/src/dsl/functions/temporal.rs:288]: This part was missing previously - processes nanosecond information.

File: crates/polars-plan/src/dsl/functions/temporal.rs [hunk 2]
```diff
@@ -256,95 +256,103 @@ impl DurationArgs {
...
-                if milliseconds.len() != max_len {
+                if is_scalar(&milliseconds) {
                     milliseconds = milliseconds.new_from_index(0, max_len);
                 }
+                if !is_zero_scalar(&nanoseconds) {
+                    milliseconds = milliseconds + (nanoseconds / 1_000_000);
+                }
+                if !is_zero_scalar(&microseconds) {
+                    milliseconds = milliseconds + (microseconds / 1_000);
+                }
```
Comment (stinodego) [inline crates/polars-plan/src/dsl/functions/temporal.rs:315]: This part was missing for handling `ms` precision.

**Review by MarcoGorelli — APPROVED** [review]
Nice! The different commits helped with the review

**Comment by stinodego** [comment]
> The different commits helped with the review

I try my best 😅 especially when mixing refactors and fixes/features.

**Pull request merged**